### PR TITLE
Stability: Cleanup Envoy/Services Package References

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Cleanup `envoy.go` and other test services

# Context

Remove `envoy.go` which contained aliases to functions/types in the envoy package. Migrate the tests to call that directly

## EnvoyInstance
A simple find/replace to convert a `services.EnvoyInstance` into an `envoy.Instance`

## EnvoyFactory
A simple find/replace to convert a `services.EnvoyFactory` into an `envoy.Factory`

## Ports
This was slightly more tricky, and I tried to document the challenge in the code. 

There are 2 sides of ports that must be kept in sync:
- The resource (gateway) must be configured with a port
- The proxy (envoy) must be run port-forwarding that port for local requests

Historically, both of these definitions were pulled from the defaults package. However, each test had to update the port dynamically `services.NextBindPort` to ensure that subsequent tests wouldn't try to use ports that were in use. 

I tried to move port management of the envoy service to be internal to the envoy package. This ensure that all tests which needed the envoy ports, would reference the envoy instance itself, instead of the defaults package. Also, the envoy factory is responsible for using different ports, so the tests are unaware. Thanks to @jbohanon for the idea!

A challenge is that our tests still use some resource fixtures, that are created from the defaults package. As a result, I needed to keep the defaults definitions for ports, and adjust them whenver we adjusted the envoy ports. To keep this complexity hidden from the users, I do this inside an internal function.

## Enterprise Impact
This code, and previous gloo PRs are all incorporated in https://github.com/solo-io/solo-projects/pull/5070. It is up to date and passing build-bot.

# Checklist:

- [X] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/main/changelogutils) which references the issue that is resolved.
- [X] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [X] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [X] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by ben-taussig-solo — COMMENTED** [review]
Looks fine, although I'm not sure I fully understand the changes made to how we handle ports. I think part of that is I didn't really have a strong understanding of how things were handled before. Additionally, how do we expect this to affect solo-projects? Will we need to make similar changes there after bumping the gloo dependency?

File: test/services/envoy/factory.go [hunk 1]
```diff
@@ -207,15 +206,24 @@ func (f *factoryImpl) NewEnvoyInstance() (*Instance, error) {
 		}
 	}
 
+	// Ensure that each Instance has a unique set of ports
```
Comment (sam-heilbron) [inline test/services/envoy/factory.go:209]: todo: clarify this better. what does it do? what does it not yet do?

File: test/services/envoy/ports.go [hunk 2]
```diff
@@ -3,22 +3,54 @@ package envoy
...
+
 	"github.com/solo-io/gloo/test/ginkgo/parallel"
 )
 
 var (
-	adminPort = uint32(20000)
-	bindPort  = uint32(10080)
+	bindPort = uint32(10080)
+
+	adminPort  = defaults.GlooAdminPort
```
Comment (sam-heilbron) [inline test/services/envoy/ports.go:14]: todo: remove duplicate variables

File: test/e2e/grpcweb_test.go [hunk 3]
```diff
@@ -87,7 +87,7 @@ var _ = Describe("Grpc Web", func() {
 		)
 
 		BeforeEach(func() {
-			accessLogPort := services.NextBindPort()
+			accessLogPort := envoy.NextBindPort()
```
Comment (ben-taussig-solo) [inline test/e2e/grpcweb_test.go:90]: Is this an oversight or an exception to the rule that individual tests shouldn't increment ports? If it is an exception, can you leave a comment explaining why this is necessary?

Comment (sam-heilbron) [reply test/e2e/grpcweb_test.go]: Good call. It was an exception to the rule, but it didn't need to be. I just pushed a commit to try to account for this.

File: test/services/envoy/ports.go [hunk 4]
```diff
@@ -9,19 +9,24 @@ import (
...
+	baseHttpPort      = defaults.HttpPort
+	baseHttpsPort     = defaults.HttpsPort
+	baseTcpPort       = defaults.TcpPort
+	baseHybridPort    = defaults.HybridPort
+	baseAccessLogPort = uint32(10080)
 )
 
-func NextBindPort() uint32 {
-	return advancePort(&bindPort)
+func NextAccessLogPort() uint32 {
```
Comment (ben-taussig-solo) [inline test/services/envoy/ports.go:20]: Does this ever get called?

Comment (sam-heilbron) [reply test/services/envoy/ports.go:20]: Yeah this is called inside the factory implementation, when a NewInstance is created

File: test/services/envoy/ports.go [hunk 5]
```diff
@@ -3,22 +3,32 @@ package envoy
...
 
+	"github.com/solo-io/gloo/projects/gloo/pkg/defaults"
+
 	"github.com/solo-io/gloo/test/ginkgo/parallel"
 )
 
 var (
-	adminPort = uint32(20000)
-	bindPort  = uint32(10080)
+	baseAdminPort     = defaults.EnvoyAdminPort
```
[the remark this answers is not in the document]
Comment (ben-taussig-solo) [reply test/services/envoy/ports.go:12]: This is fine for me -- I know that Jacob had raised this concern initially, so I'm going to leave this thread open so he has space to push back. In the meantime, I'm happy to approve the PR

**Review by ben-taussig-solo — APPROVED** [review]

File: test/services/envoy/ports.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (jbohanon) [reply test/services/envoy/ports.go:12]: I'm good with this. I think moving it here and calling it `base*Port` is more expressive.

File: test/services/nats.go
Comment (jbohanon) [inline test/services/nats.go:1]: :tada:

**Review by jbohanon — APPROVED** [review]

**Pull request merged**